lib/std/fs/File.zig

```diff
@@ -1853,7 +1853,13 @@ pub const Writer = struct {
                 return error.EndOfStream;
             }
             const consumed = io_w.consume(@intCast(sbytes));
-            file_reader.seekTo(file_reader.pos + consumed) catch return error.ReadFailed;
+            if (consumed <= file_reader.interface.bufferedLen()) {
```

Comment:
Would you mind fixing this confusing variable name while you are here? The return value of `consume()` is not the number of bytes "consumed" but rather the number left over after consuming bytes in the writer's buffer.

Perhaps `remainder` would be better? I'm having trouble thinking of other alternatives...